src/lib.rs

```diff
@@ -107,16 +107,6 @@ extern crate blas_src;
 #[cfg(feature = "blas")]
```

Comment:
I think it should be possible to remove all these other `extern crate` lines too. To check this, remove them, and then make sure that e.g. `cargo build --features="approx"` passes.